index.bs

```diff
@@ -3214,7 +3226,15 @@ The following abstract operations support the implementation of the
 
  1. Assert: |controller|.[=ReadableByteStreamController/[[pendingPullIntos]]=] is not [=list/is
     empty|empty=].
+ 1. Assert: ! [$CanTransferArrayBuffer$](|view|.\[[ViewedArrayBuffer]]) is true.
  1. Let |firstDescriptor| be |controller|.[=ReadableByteStreamController/[[pendingPullIntos]]=][0].
+ 1. Let |state| be
+    |controller|.[=ReadableByteStreamController/[[stream]]=].[=ReadableStream/[[state]]=].
+ 1. If |state| is "`closed`",
+  1. If |view|.\[[ByteLength]] is not 0, throw a {{TypeError}} exception.
+ 1. Otherwise,
+  1. Assert: |state| is "`readable`".
+  1. If |view|.\[[ByteLength]] is 0, throw a {{TypeError}} exception.
```

Comment:
These steps were moved from `RespondInternal()` to both `Respond()` and `RespondWithNewView()`. This way, we can throw errors *before* reaching this step in `RespondWithNewView()`:
> 10. Set *firstDescriptor*’s buffer to *view*.[[ViewedArrayBuffer]].

In `RespondInternal()`, these checks were turned into asserts.